gensim/models/tfidfmodel.py

```diff
@@ -6,51 +6,100 @@
 
 
 import logging
-import math
+from functools import partial
 
 from gensim import interfaces, matutils, utils
 from six import iteritems
 
+import numpy as np
 
 logger = logging.getLogger(__name__)
 
 
+def resolve_weights(smartirs):
+    """
+    Checks for validity of smartirs parameter.
+    """
+    if not isinstance(smartirs, str) or len(smartirs) != 3:
+        raise ValueError("Expected a string of length 3 except got " + smartirs)
+
+    w_tf, w_df, w_n = smartirs
+
+    if w_tf not in 'nlabL':
+        raise ValueError("Expected term frequency weight to be one of 'nlabL', except got " + w_tf)
+
+    if w_df not in 'ntp':
+        raise ValueError("Expected inverse document frequency weight to be one of 'ntp', except got " + w_df)
+
+    if w_n not in 'ncb':
+        raise ValueError("Expected normalization weight to be one of 'ncb', except got " + w_n)
+
+    return w_tf, w_df, w_n
+
+
 def df2idf(docfreq, totaldocs, log_base=2.0, add=0.0):
     """
     Compute default inverse-document-frequency for a term with document frequency `doc_freq`::
-
-      idf = add + log(totaldocs / doc_freq)
+    idf = add + log(totaldocs / doc_freq)
     """
-    return add + math.log(1.0 * totaldocs / docfreq, log_base)
+    return add + np.log(float(totaldocs) / docfreq) / np.log(log_base)
```

Comment:
What's a reason to use `np.log` instead of `math.log`?

Reply:
Consistency I am using `np` everywhere. Anyways I do'nt see any problem with that. Do you?

Reply:
No problem this is only a question :)